src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalStoreHealthExpiryHttpPipelinePolicy.cs

```diff
@@ -0,0 +1,59 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using EnsureThat;
+using Microsoft.Health.Dicom.Blob.Utilities;
+
+namespace Microsoft.Health.Dicom.Blob.Features.ExternalStore;
+
+internal class ExternalStoreHealthExpiryHttpPipelinePolicy : HttpPipelinePolicy
+{
+    private readonly ExternalBlobDataStoreConfiguration _externalStoreOptions;
+    private readonly Regex _healthCheckRegex;
+    private const string GuidRegex = @"[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}";
+
+    public ExternalStoreHealthExpiryHttpPipelinePolicy(ExternalBlobDataStoreConfiguration externalStoreOptions)
+    {
+        _externalStoreOptions = EnsureArg.IsNotNull(externalStoreOptions, nameof(externalStoreOptions));
+        EnsureArg.IsNotNull(externalStoreOptions.BlobContainerUri, nameof(externalStoreOptions.BlobContainerUri));
+        EnsureArg.IsNotNull(externalStoreOptions.StorageDirectory, nameof(externalStoreOptions.StorageDirectory));
+        EnsureArg.IsNotNull(externalStoreOptions.HealthCheckFilePath, nameof(externalStoreOptions.HealthCheckFilePath));
+
+        UriBuilder uriBuilder = new UriBuilder(_externalStoreOptions.BlobContainerUri);
+        uriBuilder.Path = Path.Combine(uriBuilder.Path, _externalStoreOptions.StorageDirectory, _externalStoreOptions.HealthCheckFilePath);
+
+        string healthCheckPathRegex = Regex.Escape(uriBuilder.Uri.AbsoluteUri);
+        string txtRegex = Regex.Escape(".txt");
+        _healthCheckRegex = new Regex($"^{healthCheckPathRegex}{GuidRegex}{txtRegex}$", RegexOptions.Compiled);
+    }
+
+    public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+    {
+        AddExpiryHeaderToHealthCheckFileUploadRequest(message);
+        ProcessNext(message, pipeline);
+    }
+
+    public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+    {
+        AddExpiryHeaderToHealthCheckFileUploadRequest(message);
+        return ProcessNextAsync(message, pipeline);
+    }
+
+    private void AddExpiryHeaderToHealthCheckFileUploadRequest(HttpMessage message)
+    {
+        if (_healthCheckRegex.IsMatch(message.Request.Uri.ToUri().AbsoluteUri) &&
+        (message.Request.Method == RequestMethod.Put || message.Request.Method == RequestMethod.Post || message.Request.Method == RequestMethod.Patch))
+        {
+            message.Request.Headers.Add("x-ms-expiry-time", $"{_externalStoreOptions.HealthCheckFileExpiry.TotalMilliseconds}");
```

Comment:
nit: invoke `ToString` with an invariant culture instead of a format string that will use current culture